tests/tests/src/lazy_test.res

```diff
@@ -52,8 +52,8 @@ let f006: Lazy.t<unit => int> = Lazy.make(() => {
   _ => x
 })
 
-let f007 = Lazy.make(() => raise(Not_found))
-let f008 = Lazy.make(() => {
+let f007: Lazy.t<string> = Lazy.make(() => raise(Not_found))
```

Comment:
Defining `lazy_t<'a>` = `Lazy.t<'a>` as an abstract type in the stdlib instead of a built-in type in predef has the consequence that one needs to specify the payload type here, otherwise one gets:

```
  This expression's type contains type variables that cannot be generalized:
```